### DISCUSSION TITLE
Using RTK-Query with AppSync?

### DISCUSSION THREAD

**Discussion by jacobgranberry** [reduxjs/redux-toolkit#2066] [Q&A] [answered]
Hi there - was wondering if anyone else has attempted to connect RTK-Query with the AppSync graphql service?  Seems its not going to work out of the box with `aws-appsync`.

Feeding the client into `graphqlRequestBaseQuery` gives me this error:

```
Type 'AWSAppSyncClient<NormalizedCacheObject>' is missing the following properties from type 'GraphQLClient': url, options, rawRequest, request, and 4 more.ts(2740)
```

I know this is probably something more on the AWS side, but just wondering if anyone else had attempted it.  

Some sample code of what I'm attempting:
```
import AppSyncClient from 'aws-appsync;

const client = new AppSyncClient({
  url: process.env.REACT_APP_COMPANYQL_URL as string,
  region: process.env.REACT_APP_REGION as string,
  auth: {
    type: 'OPENID_CONNECT',
    jwtToken: async () => MasterAuthInterface.getCore().then(auth => auth.getTokenSilently()),
  },
});

export const registrationApi = createApi({
  reducerPath: 'registrationApi',

  baseQuery: graphqlRequestBaseQuery({ client }),
  endpoints: builder => ({
    ...
  }),
});

````

**Comment by markerikson** [comment, marked as answer]
I'm not familiar with AppSync, but isn't it a GraphQL client in and of itself?  That suggests you would _not_ want to use `graphqlRequestBaseQuery`, but rather a custom base query that formats request details and passes them through to the AppSync client instance.

**Reply by jacobgranberry** [reply]
Makes sense!  Didn't even know that was an option.  Thanks for the help
